File: pset2/caesar/caesar.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[]) {
    if (argc == 2) {
        // User input and inital variables
        int k = atoi(argv[1]); // convert to int
        printf("plaintext: ");
        string s = get_string();

        if (s != NULL) {
            printf("ciphertext: ");
            // loop check string
            for (int i = 0, n = strlen(s); i < n; i++) {
                if (isalpha(s[i])) { // check whether is alphabet
                    if (isupper(s[i])) // upper alphabet # starts from 65 in ASCII
                        printf("%c", (s[i] - 65 + k) % 26 + 65);
                    else if (islower(s[i]))
                        printf("%c", (s[i] - 97 + k) % 26 + 97); // lower alphabet # starts from 97 in ASCII
                } else
                    printf("%c", s[i]); // other case
            }
            printf("\n");
            return 0;
        }
    } else {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}

File: pset6/caesar.py
import sys
import cs50

def main():
    if len(sys.argv) == 2: # argv[1:] is the argument array, len() method return the length of a string or array
        # User input and inital variables
        k = int(sys.argv[1]) # convert to int
        print("plaintext: ", end="")
        message = cs50.get_string() # get inputs

        if message != '':
            print("ciphertext: ", end="")
            # loop check string
            for i in message:
                if i.isalpha(): # check whether the character is alphabet, use .isaplha() method
                    if i.isupper(): # upper alphabet # starts from 65 in ASCII
                        print(chr((ord(i) - 65 + k) % 26 + 65), end="")
                    elif i.islower():
                        print(chr((ord(i) - 97 + k) % 26 + 97), end="") # lower alphabet # starts from 97 in ASCII
                else:
                    print(message) # other case
            print("")
            exit(0)
    else:
        print("Usage: ./caesar k\n")
        exit(1)

if __name__ == "__main__":
    main()